### ISSUE TITLE
Certificate transparency fails on switching Charles when application is running

### ISSUE THREAD

**Issue by ujjwalgoyal** [babylonhealth/certificate-transparency-android#52]
Hi,
I am using babylon 0.2 version lib. 
Steps to Repro : 
1) Open the application using charles- > Certificate transparency failes
2) Remove the Charles from app -> Still certificate transparency is shown with the following error

javax.net.ssl.SSLPeerUnverifiedException: Certificate transparency failed
     at com.babylon.certificatetransparency.internal.verifier.CertificateTransparencyInterceptor.intercept(CertificateTransparencyInterceptor.kt:59)
   at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
    at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:37)
    at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
    at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
  at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:82)
   at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
    at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
   at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:84)
    at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
  at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:71)
   at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
 at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)

Expected Result: App should work if Charles was removed when app is running. 
Note: Certificate transparency is succeeded when the app is killed and started again.

**Comment by ujjwalgoyal** [comment]
Any update on this issue?

**Comment by mattmook** [comment]
Would you be able to add the default logger and submit the output from that; there's a default implementation built-in that can be used:

```
certificateTransparencyInterceptor {
    ...
    logger = BasicAndroidCRLogger(BuildConfig.DEBUG)
}
```

**Comment by mattmook** [comment]
I've been able to get Charles to work as expected.

- On my device (an emulator) I've set the Wi-Fi proxy to point to Charles. As this is the emulator I've used 10.0.2.2 on port 8888. With 10.0.2.2 being a special address that uses the local loopback on my local machine.
- I've then installed the Charles SSL certificate.
- In the SSL settings of Charles I've added an entry for *.babylonhealth.com port 443. <img width="704" alt="Screenshot 2020-08-26 at 20 39 29" src="https://user-images.githubusercontent.com/182422/91348928-6b76f180-e7dc-11ea-9641-f14e6e8a877d.png">
- Next launching the projects sample app, I've entered the OkHttp Kotlin Example screen, selected "Test Certificate Transparency" and entered `www.babylonhealth.com`.
- This displays the expected `Too few trusted SCTs, required 2, found 0` ![Screenshot_1598470792](https://user-images.githubusercontent.com/182422/91349113-bdb81280-e7dc-11ea-88c2-162a8e3cd858.png)
- Now, back in the `Proxy > SSL Proxying Settings` in Charles, disable (or remove) the entries added previously. <img width="704" alt="Screenshot 2020-08-26 at 20 41 46" src="https://user-images.githubusercontent.com/182422/91349041-9f521700-e7dc-11ea-8cee-9ab0de45969e.png">
- Retry the sample app with `www.babylonhealth.com` and see a green Success.
![Screenshot_1598470782](https://user-images.githubusercontent.com/182422/91349182-daece100-e7dc-11ea-8f49-25e421ac55f1.png)

If instead of changing the SSL proxy settings you quit Charles, you will instead see a failure to connect to `/10.0.2.2:8888` as the proxy server is no longer running but the devices settings are still connecting to it. However, this should resolve as soon as you remove the proxy server from the Wi-Fi settings.
